docs/guides/using_TransformAnimationComponent.rst

```diff
@@ -0,0 +1,160 @@
+===================================================
+Animating Objects using TransformAnimationComponent
+===================================================
+
+TransformAnimationComponent can be used to easily animate scenes which have to be animated endlessly without relying on scripting for the same.
+
+It is quite easy to as it uses a keyframe based interface to achieve this.
+
+Setting up
+----------
+
+Create a new scene and assign the TransformComponent, ModelComponent and TransformAnimationComponent to it.
+
+Creating a new scene:
+
+.. image:: images/AnimTestCreate.png
+
+Adding the Components by right clicking on the scene in inspector:
+
+.. image:: images/AnimTestComp.png
+
+Select the EditorCamera:
+
+.. image:: images/AnimTestCam.png
+
+Set the camera to EditorCamera and position it such that it has a good view of the scene.
+
+.. image:: images/AnimTestView.png
+
+Select the EditorCamera as default camera for the scene.
+
+.. image:: images/AnimTestDefCam.png
+
+Interface
+---------
+
+* Transition Type : How the transition must be between keyframes. Explaining the options:
+
+  - SmashSmash : Abrupt start and end transition
+  - EaseEase : Smooth start and end transition
+  - SmashEase : Abrubt start and smooth end transition
+  - EaseSmash : Smooth start and abrupt end transition
+
+* Animation Mode : How the animation should play. Explaining the options:
+
+  - Looping : Animation plays from Start time to End time unidirectionally.
+  - Alternating : Animation plays from Start time to EndTime and then reverses from End time to Start time.
+
+* Time : Shows the time while animation plays
+
+* Reset : resets the animation
+
+* Play in Editor : Play the animation in editor
+
+* Play on start : Play the animation when game starts
+
+* Keyframes : To set the animation keyframes
+
+  - Set Keyframe : To add a new keyframe at the end
+  - Pop Keyframe : To remove the last keyframe
+
+Transition Type
+---------------
+
+Examples of different transition types for better understanding.
+
+SmashSmash:
+
+.. image:: images/SmashSmash.gif
+
+EaseEase:
+
+.. image:: images/EaseEase.gif
+
+SmashEase:
+
+.. image:: images/SmashEase.gif
+
+EaseSmash:
+
+.. image:: images/EaseSmash.gif
+
+Animation Mode
+--------------
+
+Showing demo of each mode for better understanding.
+
+None: Animation plays only once.
+
+.. image:: images/AnimNone.gif
+
+Looping: Animation plays unidirectionally and repeats afteer ending.
+
+.. image:: images/AnimLoop.gif
+
+Alternating: Animation plays back and forth (bidirectionally).
+
+.. image:: images/AnimAlt.gif
+
+Check the time progressbar for more clarity.
+
+Reset
+-----
+
+Resets the animation to the starting frame (Time 0.00).
+
+.. image:: images/AnimReset.gif
+
+Play in Editor
+--------------
+
+Plays the Animation in EditorView.
+
+.. image:: images/AnimEditor.gif
+
+Play on Start
+-------------
+
+Plays the Animation on game start.
+
+.. image:: images/AnimGame.gif
+
+Keyframes
```

Comment:
I think keyframes section should be moved up so the reader has an idea about how to set up animation before seeing examples of what all the options. Move this to after the interface section.